client/src/graphql_utils/graphql_utils.js

```diff
@@ -97,70 +106,98 @@ export function get_client() {
   return client;
 }
 
-export const query_logging_wrapper = (
-  query_name,
-  query,
-  expect_resolved_response = true
-) => (variables) => {
+const make_query_promise = (query_name, query, resolver) => (variables) => {
   const time_at_request = Date.now();
-  return query({ ...variables, _query_name: query_name })
+
+  return get_client()
+    .query({
+      query: query,
+      variables: {
+        ...variables,
+        _query_name: query_name,
+      },
+    })
+    .then(resolver)
     .then((resolved_response) => {
-      const resp_time = Date.now() - time_at_request;
-      if (!expect_resolved_response || !_.isEmpty(resolved_response)) {
-        // Not a very good test, might report success with unexpected data... ah well, that's the API's job to test!
-        log_standard_event({
-          SUBAPP: window.location.hash.replace("#", ""),
-          MISC1: "API_QUERY_SUCCESS",
-          MISC2: `${query_name}, took ${resp_time} ms`,
-        });
-      } else {
-        log_standard_event({
-          SUBAPP: window.location.hash.replace("#", ""),
-          MISC1: "API_QUERY_UNEXPECTED",
-          MISC2: `${query_name}, took ${resp_time} ms`,
-        });
-      }
+      log_standard_event({
+        SUBAPP: window.location.hash.replace("#", ""),
+        MISC1: "API_QUERY_SUCCESS",
+        MISC2: `${query_name}, took ${Date.now() - time_at_request} ms`,
+      });
 
       return resolved_response;
     })
     .catch((error) => {
-      const resp_time = Date.now() - time_at_request;
       log_standard_event({
         SUBAPP: window.location.hash.replace("#", ""),
         MISC1: "API_QUERY_FAILURE",
-        MISC2: `${query_name}, took ${resp_time} ms - ${error.toString()}`,
+        MISC2: `${query_name}, took ${
+          Date.now() - time_at_request
+        } ms - ${error.toString()}`,
       });
+
       throw error;
     });
 };
 
-const InnerLoadingHoc = ({ Component, data_to_props }) => (props) => {
-  if (props.data.loading) {
-    return <div> Loading ... </div>;
+const make_query_hook = (query_name, query, resolver) => (variables) => {
```

Comment:
Hooks should always follow the `use*` naming pattern, otherwise it's not clear that this is indeed a hook